src/borg/remote.py

```diff
@@ -646,12 +654,27 @@ def borg_cmd(self, args, testing):
                 opts.append('--critical')
             else:
                 raise ValueError('log level missing, fix this code')
-            try:
-                borg_logger = logging.getLogger('borg')
-                if borg_logger.json:
-                    opts.append('--log-json')
-            except AttributeError:
-                pass
+
+            # Tell the remote server about debug topics it may need to consider.
+            # Note that debug topics are usable for "spew" or "trace" logs which would
+            # be too plentiful to transfer for normal use, so the server doesn't send
+            # them unless explicitly enabled.
+            #
+            # Needless to say, if you do --debug-topic=repository.compaction, for example,
+            # with a 1.0.x server it won't work, because the server does not recognize the
+            # option.
+            #
+            # This is not considered a problem, since this is a debugging feature that
+            # should not be used for regular use.
+            for topic in args.debug_topics:
+                # Check whether either the unqualified ...
+                concerns_repository = topic.startswith('repository')
+                if '.' not in topic:
+                    topic = 'borg.debug.' + topic
+                # ... or the qualified topic starts with repository.
+                concerns_repository |= topic.startswith('repository')
```

Comment:
It could be `repository.xy`, but it's still wrong, since we'd use something like `borg.repository[xxx]`. So, rather just `repository in topic`.